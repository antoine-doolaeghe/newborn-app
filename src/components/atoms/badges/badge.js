import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Theme } from "../../../theme/theme";

export const StyledBadge = styled.div`
  align-items: center;
  background-color: ${Theme.palette.success};
  border-radius: ${Theme.radius.small};
  color: white;
  display: flex;
  font-family: ${Theme.fontFamily};
  font-size: ${Theme.fontSize.small};
  font-weight: ${Theme.weight.normal};
  font-weight: ${Theme.weight.normal};
  height: ${props => props.height};
  justify-content: center;
  padding: 0px ${Theme.spacing.small};
  margin: ${Theme.spacing.small};
  width: ${props => props.width};
`;

const Badge = ({ label, width, height }) => {
  return (
    <StyledBadge width={width} height={height}>
      {label}
    </StyledBadge>
  );
};

Badge.defaultProps = {
  height: "25px",
  width: "auto",
  label: "--"
};

Badge.propTypes = {
  width: PropTypes.string,
  height: PropTypes.string,
  label: PropTypes.string
};

export default Badge;
