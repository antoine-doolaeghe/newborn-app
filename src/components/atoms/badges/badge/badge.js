import React from "react";
import styled from "styled-components";

export const StyledBadge = styled.div`
  background-color: lightgreen;
  border-radius: 2px;
  display: flex;
  justify-content: center;
  font-family: "IBM Plex Sans", sans-serif;
  font-size: 0.875rem;
  font-weight: 400;
  align-items: center;
  font-size: 15px;
  font-weight: 400;
  height: 25px;
  width: 35px;
  margin: 5px;
`;

const Badge = ({ label }) => {
  return <StyledBadge>{!label ? "--" : label}</StyledBadge>;
};
export default Badge;
