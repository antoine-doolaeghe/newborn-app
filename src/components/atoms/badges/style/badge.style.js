import styled from "styled-components";
import { Theme } from "../../../../theme/theme";

export const StyledBadge = styled.div`
  align-items: center;
  background-color: ${Theme.palette.success};
  border-radius: ${Theme.radius.small};
  color: white;
  display: flex;
  font-family: ${Theme.fontFamily};
  font-size: ${Theme.fontSize.small};
  font-weight: ${Theme.weight.normal};
  font-weight: ${Theme.weight.normal};
  height: ${props => props.height};
  justify-content: center;
  padding: 0px ${Theme.spacing.small};
  margin: ${Theme.spacing.small};
  width: ${props => props.width};
`;
