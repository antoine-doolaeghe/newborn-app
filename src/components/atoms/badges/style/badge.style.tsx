import React from "react";
import styled from "styled-components";
import { Theme } from "../../../../theme/theme";

interface IStyledBadgeProps {
  width: string;
  height: string;
  children: string;
}

export const StyledBadge = styled(
  (props: IStyledBadgeProps): React.ReactElement => <div {...props} />
)((props): string => {
  return `
    align-items: center;
    background-color: ${Theme.palette.success};
    border-radius: ${Theme.radius.small};
    color: white;
    display: flex;
    font-family: ${Theme.fontFamily};
    font-size: ${Theme.fontSize.small};
    font-weight: ${Theme.weight.normal};
    font-weight: ${Theme.weight.normal};
    height: ${props.height};
    justify-content: center;
    padding: 0px ${Theme.spacing.small};
    margin: ${Theme.spacing.small};
    width: ${props.width};
  `;
});
