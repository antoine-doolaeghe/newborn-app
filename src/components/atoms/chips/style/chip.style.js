import styled from "styled-components";
import Chip from "@material-ui/core/Chip";
import { Theme } from "../../../../theme/theme";

const { spacing } = Theme;

export const StyledChip = styled(Chip)`
  margin: ${spacing.small};
`;
