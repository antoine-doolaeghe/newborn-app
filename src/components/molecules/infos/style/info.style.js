import styled, { keyframes } from "styled-components";

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

export const InfoElement = styled.section`
  width: 105px;
`;

export const InfoWrap = styled.div`
  display: flex;
  flex: 1;
  margin-right: 10px;
`;

export const Label = styled.div`
  font-size: 12px;
  font-weight: 300;
  margin-top: 5px;
  text-transform: uppercase;
`;

export const Value = styled.div`
  font-size: ${props => (props.fontSize ? props.fontSize : "18px")};
  font-weight: 600;
  text-transform: capitalize;
`;

export const InfoLabelIcon = styled.div`
  margin: 10px 10px 10px 0px;
  width: 55px;
  height: 55px;
  border-radius: 40px;
  font-size: 30px;
  color: white;
  background-color: lightgreen;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const InfoLabelAvatar = styled.div`
  margin: 10px 10px 10px 0px;
  width: 55px;
  height: 55px;
  cursor: pointer;
  border-radius: 40px;
  font-size: 30px;
  color: white;
  background-image: url(${props => props.avatar});
  background-size: 125%;
  display: flex;
  justify-content: center;
  align-items: center;
  /* border: 3px solid white; */
`;

export const InfoLabelAvatarBorder = styled.div`
  border: 2px dashed black;
  border-radius: 40px;
  position: absolute;
  width: 64px;
  height: 64px;
  &:hover {
    border: 2px dashed red;
    animation: ${rotate} 7s linear infinite;
  }
`;
