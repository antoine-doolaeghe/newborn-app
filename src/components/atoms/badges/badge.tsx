import React from "react";
import { StyledBadge } from "./style/badge.style";

export interface IBadgeProps {
  label: string;
  width: string;
  height: string;
}

const DefaultProps = {
  height: "25px",
  width: "auto",
  label: "--"
};

const Badge = ({ width, height, label }: IBadgeProps): React.ReactElement => {
  return (
    <StyledBadge width={width} height={height}>
      {label}
    </StyledBadge>
  );
};

Badge.defaultProps = DefaultProps;

export default Badge;
