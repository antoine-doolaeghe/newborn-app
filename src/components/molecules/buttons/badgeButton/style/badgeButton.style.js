import styled from "styled-components";
import { Theme } from "../../../../../theme/theme";
import { StyledDefaultButton } from "../../defaultButton/style/defaultButton.style";

export const StyledBadgeButton = styled(StyledDefaultButton)`
  align-items: center;
  background: ${props => Theme.palette[props.color].main};
  border-radius: ${Theme.radius.small};
  border: none;
  color: ${Theme.font.light};
  display: flex;
  font-family: ${Theme.fontFamily};
  font-size: ${Theme.fontSize.xsmall};
  font-weight: ${Theme.weight.normal};
  justify-content: space-around;
  margin: ${Theme.spacing.small};
  outline: none;
  padding: 3px 7px;
  width: fit-content;
`;
