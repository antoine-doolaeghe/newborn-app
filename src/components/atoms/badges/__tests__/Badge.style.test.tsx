import { StyledBadge } from "../style/badge.style";

describe("Badge Component Style", () => {
  it("matches the component style snapshots", () => {
    expect(StyledBadge).toMatchSnapshot();
  });
});
