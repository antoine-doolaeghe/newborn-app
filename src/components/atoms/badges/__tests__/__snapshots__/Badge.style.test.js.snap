// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component Style matches the component style snapshots 1`] = `
Object {
  "$$typeof": Symbol(react.forward_ref),
  "attrs": Array [],
  "componentStyle": ComponentStyle {
    "componentId": "sc-bdVaJa",
    "isStatic": false,
    "rules": Array [
      "
  align-items: center;
  background-color: ",
      "#00b894",
      ";
  border-radius: ",
      "5px",
      ";
  color: white;
  display: flex;
  font-family: ",
      "IBM Plex Sans",
      ";
  font-size: ",
      "1rem",
      ";
  font-weight: ",
      "400",
      ";
  font-weight: ",
      "400",
      ";
  height: ",
      [Function],
      ";
  justify-content: center;
  padding: 0px ",
      "5px",
      ";
  margin: ",
      "5px",
      ";
  width: ",
      [Function],
      ";
",
    ],
  },
  "displayName": "styled.div",
  "foldedComponentIds": Array [],
  "render": [Function],
  "styledComponentId": "sc-bdVaJa",
  "target": "div",
  "toString": [Function],
  "warnTooManyClasses": [Function],
  "withComponent": [Function],
}
`;
